Update to buster

alexgg opened the pull request:
* balenafin-firmware: Update balenafin overlay
* sd887-mrvl: Update to tag 0.0.5, maintainers and fix for Buster kernels
* Update to buster and add support to build module-only packages

alexgg committed:
sd887-mrvl: Update to tag 0.0.6

This new version contains a fix for the Buster kernel version.

Change-type: minor
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg committed:
balenafin-firmware: Update balenafin overlay

Update to support Buster.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg commented on Dockerfile:
```diff
@@ -1,5 +1,14 @@
 RUN apt-get update
 RUN apt-get -y install coreutils quilt parted debootstrap zerofree pxz zip dosfstools bsdtar libcap2-bin grep rsync xz-utils file git curl kmod fakeroot device-tree-compiler dkms
-RUN apt-get -y install build-essential dh-make
+RUN apt-get -y install build-essential dh-make raspberrypi-kernel-headers bc sudo python bison flex libssl-dev
```
Review whether all these dependencies are needed now that we are not rebuilding any in-tree module.

alexgg committed:
sd8887-mrvl-firmware: Add firmware only package

The sd8997-mrvl package is a DKMS package that includes all driver source
and firmware. However, the driver source cannot be distributed.

sd8887-mrvl-modules packages are created from the sd8887-mrvl package to
contain only the built kernel modules.

A separate sd8887-mrvl-firmware package is introduced here to contain the
firmware and configuration files complimentary to the sd8887-mrvl-modules
packages.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg committed:
balenafin-ify.sh: Update to separate sd8887 packages in binary modules and firmware.

The SD8887 driver is now installed in two parts, a sd8887-mrvl-modules
package for the driver binary modules, and a sd8887-mrvl-firmware package
for the firmware and configuration files.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg committed:
README: Update instructions

Add instructions to build binary module-only packages.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg committed:
upload-to-bintray: Fix arguments

Specify the debian package to upload with a `-d` and not a `-f` to make it
coherent with the documentation.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg committed:
gen-deb-container: Add a no-cache option

This is useful for CI as we want to always use the latest Raspbian and
not a cached version.

Signed-off-by: Alex Gonzalez <alexg@balena.io>

chriswoj commented on balenafin-ify.sh:
```diff
@@ -65,13 +67,19 @@ log "Importing bintray gpg key..."
 
 log "Installing required packages..."
 apt-get -y install apt-transport-https
 apt-get -y update
```
On a vanilla raspbian OS the apt-get update should be done before the first install. 

```
pi@raspberrypi:~/balenafin-raspbian $ sudo ./balenafin-ify.sh
[LOG][balenafin-ify.sh] Importing bintray gpg key...
OK
[LOG][balenafin-ify.sh] Adding balenaFin Raspbian repository...
[LOG][balenafin-ify.sh] Installing required packages...
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following NEW packages will be installed:
  apt-transport-https
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 149 kB of archives.
After this operation, 156 kB of additional disk space will be used.
Err:1 http://raspbian.raspberrypi.org/raspbian buster/main armhf apt-transport-https all 1.8.2.1
  404  Not Found [IP: 93.93.128.193 80]
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/a/apt/apt-transport-https_1.8.2.1_all.deb  404  Not Found [IP: 93                                                                                                                                                                                      .93.128.193 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
```

alexgg replied:
Thanks for the feedback, I have now updated the PR.

alexgg committed:
balenafin-ify: Update sources before installing packages

The package index needs to be synchronized before package installation.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg commented:
@balena-ci retest

alexgg commented:
@balena-ci retest

alexgg commented:
@balena-ci retest

alexgg committed:
Parametrize distribution and update to Buster

Add a distro-config include to parametrize the distribution version.

Change-type: minor
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg committed:
Add support to build module-only packages

On occasions the source code for a kernel module cannot be distributed
and we need to build kernel version specific packages that contain just
the kernel modules.

Building modules-only packages require to use sudo, so this is added to
the builder container.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>

nazrhom commented:
@balena-ci retest
